lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1065,6 +1062,18 @@ def loki_push_api(self):
         """Fetch Loki Push API endpoint sent from LokiPushApiProvider through relation data.
 
         Returns:
-            Loki Push API endpoint
+            The Loki Push API endpoint (or None) if the relation this `LokiPushApiConsumer` tracks
+            has limit 1; with any other relation limit, it returns a list of Loki Push API
+            endpoints, which may be empty in case there are no relation instances.
         """
-        return self._stored.loki_push_api
+        loki_endpoints = [  # Convert from the StoredSet data structure to a plain list``
+            loki_endpoint for loki_endpoint in self._stored.loki_push_api.values()
+        ]
+
+        if self._is_multi:
+            return loki_endpoints
+        else:
+            if len(loki_endpoints) == 1:
+                return loki_endpoints[0]
+            else:
+                return None
```

Comment:
This property may return:
- A list
- A string
- None

Maybe a better approach will be a property that always returns a **list**:

- A list with many elements
- A list with one element
- An empty list